### PR TITLE
Collections API improvements

### PR DESCRIPTION
**Pull request by UladBohdan**
* collections-related bugs fixed
* collection articles now have `collectionPrev` and `collectionNext` fields
* improved articles/collections serialization
* no unpublished articles returned as a part of a collection unless user has proper permissions
* collections of articles can now be updated through Bundled API
* tested.
AND
* fetching diaries from google spreadsheet is implemented.

### CODE REVIEW

**Comment by UladBohdan** [comment]
[API: return next/prev items of a collection together with the article](https://trello.com/c/49OqDVPH/201-api-return-next-prev-items-of-a-collection-together-with-the-article)

**Comment by UladBohdan** [comment]
[avoid 'export'-ing in npm scripts to support Windows properly](https://trello.com/c/Gv4sHOEv/99-avoid-export-ing-in-npm-scripts-to-support-windows-properly)

File: src/api/article/article.model.js [hunk 1]
```diff
@@ -58,9 +58,11 @@ const ArticleSchema = new Schema(
 
 const Article = mongoose.model('Article', ArticleSchema);
 
-export const serializeArticle = article => {
+// includeCollection flag here is to be able to avoid including collections
+// in case we're serializing an article into the ArticleCollection object..
+export const serializeArticle = (article, includeCollection = true) => {
```
Comment (Drapegnik) [inline src/api/article/article.model.js]: better use object as second param: `{ includeCollection = true }`,

compare:
```js
serializeArticle(article, false);
serializeArticle(article, { includeCollection: false });
```

File: src/api/article/article.model.js [hunk 2]
```diff
@@ -85,7 +87,9 @@ export const serializeArticle = article => {
   }
   return {
     ...omit(article.toObject(), ['__v', 'collectionId']),
-    collection: article.collectionId && omit(article.collectionId.toObject(), ['articles']),
+    collection: includeCollection
+      ? article.collectionId && omit(article.collectionId.toObject(), ['articles'])
```
Comment (Drapegnik) [inline src/api/article/article.model.js]: you can omit ternary:
```js

...
if (includeArticle) {
	result.collection = article.collectionId && ...
}

```

File: src/api/article/collection/controller.js [hunk 3]
```diff
@@ -1,29 +1,21 @@
...
+
+export const getAll = ({ user }, res, next) =>
   ArticleCollection.find({ active: true })
-    .populate({
-      path: 'articles',
-      populate: {
-        path: 'locales',
-      },
-    })
+    .populate(COLLECTION_POPULATE_OPTIONS.articles(!checkPermissions(user, ['canManageArticles'])))
```
Comment (Drapegnik) [inline src/api/article/collection/controller.js]: imho we can refactor `checkPermissions` to support both: array & string as second param

File: src/api/article/article.model.js [hunk 4]
```diff
@@ -107,15 +111,16 @@ export const POPULATE_OPTIONS = {
   // TODO(uladbohdan): to merge with User basicFields.
   author: '-_id firstName lastName email role active bio imageUrl displayName',
   brand: '-_id slug names imageUrl imageUrlSmall',
-  collection: {
+  collection: publishedOnly => ({
     path: 'collectionId',
     select: '-_id name slug description imageUrl articles',
     populate: {
       path: 'articles',
+      match: publishedOnly ? { publishAt: { $lt: Date.now() } } : undefined,
```
Comment (Drapegnik) [inline src/api/article/article.model.js]: looks ugly

File: src/api/article/collection/tests.js [hunk 5]
```diff
@@ -29,10 +29,10 @@ describe('Collections API', () => {
     }
     await Promise.all(articlePromises);
 
-    const articlesDict = {};
+    const articlesList = [];
     const articles = await Article.find().exec();
-    await articles.forEach(item => {
-      articlesDict[item.slug] = item._id;
+    articles.forEach(item => {
```
Comment (Drapegnik) [inline src/api/article/collection/tests.js]:
```js
const articlesList = articles.map(({ _id }) => _id));
```

Comment (UladBohdan) [reply src/api/article/collection/tests.js]: 👍

File: src/api/article/article.model.js (hunk 4 above)
Comment (UladBohdan) [reply src/api/article/article.model.js]: it does. But that incapsulates all the logic inside of the only object, that's a nice thing

File: src/api/article/article.model.js (hunk 2 above)
Comment (UladBohdan) [reply src/api/article/article.model.js]: done.

File: src/api/article/article.model.js (hunk 1 above)
Comment (UladBohdan) [reply src/api/article/article.model.js]: That's a nice approach!

File: src/api/article/collection/controller.js (hunk 3 above)
Comment (UladBohdan) [reply src/api/article/collection/controller.js]: refactored that, although the only benefit - to be able to avoid `[ ]`

**Pull request merged**